server/api/region.go

```diff
@@ -262,14 +262,20 @@ func (h *regionsHandler) CheckRegionsReplicated(w http.ResponseWriter, r *http.R
 	}
 
 	regions := rc.ScanRegions(startKey, endKey, -1)
-	replicated := true
+	state := "REPLICATED"
 	for _, region := range regions {
 		if !opt.IsRegionReplicated(rc, region) {
-			replicated = false
+			state = "INPROGRESS"
+			for _, item := range rc.GetCoordinator().GetWaitingRegions() {
```

Comment:
What is the difference between `INPROGRESS` and `WAITING`. Sound like they have a similar meaning. BTW, waiting regions is a cache that may be empty sometimes.

Reply:
`INPROGRESS` means there is an operator executing, while `WAITING` basically means the scheduling stalls somehow. If you have a better name, I'd like to replace them.

Reply:
It does not really matter if it is empty. This is not so strict anyway as long as it is expected under good network and load condition.

Reply:
It is changed to "PENDING" according to morgo, https://github.com/pingcap/tidb/pull/31073#issuecomment-1002610790